plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1375,6 +1375,25 @@ public void testCheckpointing()
         assertThat(listCheckpointFiles(transactionLogDirectory)).hasSize(2);
         assertQuery("SELECT * FROM " + tableName + " WHERE a_string <> 'fill'", "VALUES (1,'ala'),  (3,'psa'), (2, 'bobra')");
 
+        // replace table
+        assertUpdate(
+                format("CREATE OR REPLACE TABLE %s (a_number) " +
+                                " WITH (location = '%s', " +
+                                "       checkpoint_interval = 5) " +
+                                " AS VALUES (1), (2), (3), (4), (5)",
+                        tableName,
+                        getLocationForTable(bucketName, tableName)),
+                5);
+
+        // sanity check
+        assertThat(listCheckpointFiles(transactionLogDirectory)).hasSize(2);
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1), (2), (3), (4), (5)");
+
+        // fill to another checkpoint which reads previous checkpoint
+        fillWithInserts(tableName, "(1)", 5);
+        assertThat(listCheckpointFiles(transactionLogDirectory)).hasSize(3);
+        assertQuery("SELECT COUNT(*) FROM " + tableName, "VALUES (10)");
```

Comment:
why is this relevant?

Reply:
I added this assertion in order to comply with [this comment](https://github.com/trinodb/trino/pull/21609#discussion_r1572537509):

> We also need an accuracy test in TestDeltaLakeConnectorTest to verify whether the table is still readable after this scenario.

What type of accuracy test did you have in mind?

Reply:
The following statement reads the checkpoint as well

```
assertQuery("SELECT * FROM " + tableName, "VALUES (1), (2), (3), (4), (5)");
```

(btw `assertQuery("SELECT * FROM " + tableName, "VALUES 1, 2, 3, 4, 5");` should work as well)